spring-integration-core/src/main/java/org/springframework/integration/annotation/Role.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotate endpoints to assign them to a role. Such endpoints can be started/stopped as
+ * a group. See {@code SmartLifecycleRoleController}.
+ *
+ * @author Gary Russell
+ * @since 4.2
+ */
+@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@Documented
+public @interface Role {
```

Comment:
Since it reflects `AbstractLeaderEvent.getRole()`, it really seems for me like `@LeadershipRole` is a good name.
Or just `@Leadership`. Even if it is a `role`from the `Candidate` it means that our lifecycles are "guided"...
Need to think more, or just leave as `@Role` and make other to live with that! :smile: